### PR TITLE
[RISCV][TTI] Add llvm.vp.select into canSplatOperand.

### PR DESCRIPTION
**Pull request by LiqinWeng**
The second operand of llvm.vp.select is a splat operand , it can help llvm.vp.select  fold vv instructions to vx instructions.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ; RUN: llc < %s -mtriple=riscv64 -mattr=+m,+v,+f -target-abi=lp64f \
 ; RUN:     | FileCheck %s
 
-define void @sink_splat_mul(ptr nocapture %a, i32 signext %x) {
+define void @sink_splat_mul(ptr %a, i32 %x) {
```
Comment (lukel97) [inline llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll:5]: Do we need to change these existing tests?

Comment (LiqinWeng) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll:5]: It seems that this information is unnecessary, so I deleted it.

Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll:5]: Can you split that out into a separate NFC PR just so we can make sure it isn't affecting this diff?

File: llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll [hunk 2]
```diff
@@ -22,7 +22,7 @@ entry:
   %broadcast.splat = shufflevector <4 x i32> %broadcast.splatinsert, <4 x i32> poison, <4 x i32> zeroinitializer
   br label %vector.body
 
-vector.body:                                      ; preds = %vector.body, %entry
+vector.body:
```
Comment (lukel97) [inline llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll:25]: Same here for these comments, I think these are obscuring the actual code related changes from the diff

Comment (LiqinWeng) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll]: you nean delete it or not?

File: llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll (hunk 1 above)
Comment (LiqinWeng) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll]: OK, fixed

Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll]: Thanks. Could you also separate out the new tests in this PR into a separate commit so we can see the diff? Feel free to force push if you need

Comment (LiqinWeng) [reply llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll]: done

**Review by lukel97 — APPROVED** [review]

**Comment by lukel97** [comment]
Thanks. Maybe add a bit in the PR description to mention that this allows us to select vmerge.vxm?

**Pull request merged**